### PR TITLE
v0.0.1

### PR DESCRIPTION
**Pull request by yotamloe**
First release of the `EZkonnect-server` :
- API endpoints: `[GET] /api/v1/state`, `[POST] /api/v1/annotate/traces`, `[POST] /api/v1/annotate/logs`.
- Docker file and Kubernetes manifests.

The `dev` [branch](https://github.com/logzio/ezkonnect-server/tree/dev) contains a working version, we can start work with it and make changes as we go.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+IMAGE_NAME := ezkonnect-server
+IMAGE_TAG := 0.0.1
+DOCKER_REPO := yotamloe/$(IMAGE_NAME):$(IMAGE_TAG)
```
Comment (mirii1994) [inline Makefile:3]: Your repo :)

File: api/annotate/traces.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		if !isValidTracesResourceRequest(resource) {
+			return false
+		}
+	}
+	return true
+}
+
+func isValidTracesResourceRequest(req TracesResourceRequest) bool {
+	isValidKind := req.Kind == "statefulset" || req.Kind == "deployment"
+	isValidAction := req.Action == "add" || req.Action == "delete"
```
Comment (mirii1994) [inline api/annotate/traces.go:158]: I'd prefer if:
1. we'll move these to be consts (maybe even under common.go)
2. In the function create an array of supported kinds and supported actions.
3. Iterate through those arrays to check validity
4. Use those consts throughout the code

File: api/annotate/traces.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	if !validRequests {
+		logger.Error("Invalid input")
+		http.Error(w, "Invalid input ", http.StatusBadRequest)
+		return
+	}
+
+	var responses []TracesResourceResponse
+	for _, resource := range resources {
+		// choose the annotation key and value according to the telemetry type and action
+		var annotationKey = "logz.io/traces_instrument"
```
Comment (mirii1994) [inline api/annotate/traces.go:73]: Please move it to be a const and use it throughout the code

File: api/annotate/traces.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	if err != nil {
+		logger.Error("Error decoding JSON body", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	// Get the Kubernetes config
+	config, err := api.GetConfig()
+	if err != nil {
+		logger.Error("Error getting Kubernetes config", err)
+		http.Error(w, "Error getting Kubernetes config", http.StatusInternalServerError)
```
Comment (mirii1994) [inline api/annotate/traces.go:51]: For maintenance purposes - would it be better if we'd have the message in a variable, and refer to it in both logger & http error?

File: api/annotate/traces.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+	"github.com/logzio/ezkonnect-server/api"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"net/http"
+)
+
+// TracesResourceRequest ResourceRequest is the JSON body of the POST request
+// It contains the name, kind, namespace, telemetry type and action of the resource
+// name: name of the resource
+// kind: kind of the resource (deployment or statefulset)
```
Comment (mirii1994) [inline api/annotate/traces.go:14]: Refer to where the consts are for supported kinds

File: api/annotate/traces.go [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+// namespace: namespace of the resource
+// action: action to perform (add or delete)
```
Comment (mirii1994) [inline api/annotate/traces.go:16]: Refer to where the consts are for supported actions

**Review by mirii1994 — CHANGES_REQUESTED** [review]
Great work!
Please see my comments :)

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,24 @@
+IMAGE_NAME := ezkonnect-server
+IMAGE_TAG := 0.0.1
```
[the remark this answers is not in the document]
Comment (yotamloe) [reply Makefile:2]: Sure I know these workflows, just didn't got to that yet :)

**Review by mirii1994 — APPROVED** [review]
YAAAAS.

**Pull request merged**